pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,258 @@
+                setPublicKey('');
+            }
+        }
+
+        getPublicKey();
+    }, [privateKey]);
+
+    function onSubmit() {
+        // Validate Addresses before submit
+        let addr;
+        try {
+            addr = stringToAddresses(addresses);
+        } catch (e) {
+            setDialogError(e.message);
+            return;
+        }
+
+        function createSettingsObj() {
+            return {
+                ...settings,
+                connection: {
+                    ...settings.connection,
+                    id: `con-${iface}`,
+                    interface_name: iface,
+                    type: 'wireguard'
+                },
+                wireguard: {
+                    private_key: privateKey.trim(),
+                    listen_port: +listenPort,
+                    peers: peers.map(peer => ({ ...peer, allowedIps: peer.allowedIps.split(',') })),
+                },
+                ipv4: {
+                    addresses: addr,
+                    method: 'manual',
+                    dns: [],
+                    dns_search: []
+                }
+            };
+        }
+
+        dialogSave({
+            connection,
+            dev,
+            model,
+            settings: createSettingsObj(),
+            onClose: Dialogs.close,
+            setDialogError
+        });
+    }
+
+    return (
+        <NetworkModal title={_("WireGuard settings")} onSubmit={onSubmit} dialogError={dialogError} idPrefix={idPrefix} submitDisabled={!iface || !addresses || !privateKey}>
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_("Private key")} fieldId={idPrefix + '-private-key-input'}>
+                <InputGroup>
+                    <TextInput id={idPrefix + '-private-key-input'} className='pf-v5-u-font-family-monospace' value={privateKey} onChange={(_, val) => setPrivateKey(val)} />
+                    <Button variant='control'
+                        onClick={async () => {
+                            try {
+                                const key = await cockpit.spawn(["wg", "genkey"]);
+                                setPrivateKey(key.trim());
+                            } catch (e) {
+                                setDialogError(e.message);
+                            }
+                        }}
+                    >
+                        {_("Generate")}
+                    </Button>
+                </InputGroup>
+            </FormGroup>
+            <FormGroup label={_("Public key")}>
+                <ClipboardCopy isReadOnly className='pf-v5-u-font-family-monospace'>{publicKey}</ClipboardCopy>
+            </FormGroup>
+            <FormGroup label={_("Listen port")} fieldId={idPrefix + '-listen-port-input'}>
+                <TextInput id={idPrefix + '-listen-port-input'} placeholder={_("Random port selected when unspecified")} type='number' value={listenPort} onChange={(_, val) => { setListenPort(val) }} />
```

Comment:
Please add validation to this and a comment that points to https://github.com/patternfly/patternfly-react/issues/9391 . The current behaviour of interpreting anything non-numeric as "0" and succeeding is very confusing.